AlicerceTests/ServiceLocatorTests.swift

```diff
@@ -189,11 +383,129 @@ class ServiceLocatorTests : XCTestCase {
             XCTAssertEqual(rStringService, testLazyServiceString())
             
         } catch let error {
-            XCTAssert(false, "💥: Register failed with error: \(error)")
+            XCTFail("💥: Get failed with error: \(error)")
+        }
+    }
+
+    // MARK: Failure
+
+    func testGet_UsingTypeInference_ShouldFailWithInexistingService() {
+
+        do {
+
+            // Try to fetch any service. Shouldn't return nothing.
+            let _: String = try serviceLocator.get()
+
+        } catch  ServiceLocatorError.inexistentService {
+            // Expected result
+        } catch {
+            XCTFail("💥: Get failed with error: \(error)")
+        }
+    }
+
+    func testGet_UsingName_ShouldFailWithInexistingService() {
+
+        let serviceName = "TestServiceName"
+
+        do {
+
+            // Try to fetch any service. Shouldn't return nothing.
+            let _: String = try serviceLocator.get(name: serviceName)
+
+        } catch  ServiceLocatorError.inexistentService {
+            // Expected result
+        } catch {
+            XCTFail("💥: Get failed with error: \(error)")
+        }
+    }
+
+    func testGet_UsingTypeInference_ShouldFailWithServiceTypeMismatch() {
+
+        let testService = ""
+        let serviceName = "Double"
+
+        do {
+
+            // Register a String type service
+            try serviceLocator.register(name: serviceName, service: testService)
+
+            // Get a Double type service
+            let _: Double = try serviceLocator.get()
+
+        } catch ServiceLocatorError.serviceTypeMismatch(expected: let expectedType, found: let foundType) {
+            XCTAssertTrue(type(of: expectedType) != type(of: foundType))
+        } catch {
+            XCTFail("💥: Get failed with error: \(error)")
+        }
+    }
+
+    func testGet_UsingName_ShouldFailWithServiceTypeMismatch() {
+
+        let testService = ""
+        let serviceName = "TestServiceName"
+
+        do {
+
+            // Register a String type service
+            try serviceLocator.register(name: serviceName, service: testService)
+
+            // Get a Double type service
+            let _: Double = try serviceLocator.get(name: serviceName)
+
+        } catch ServiceLocatorError.serviceTypeMismatch(expected: let expectedType, found: let foundType) {
+            XCTAssertTrue(type(of: foundType) != type(of: expectedType))
```

Comment:
regarding the test with double, i'm not sure if i agree with this one. We could besides that validate that the type is not the same as the `testService`